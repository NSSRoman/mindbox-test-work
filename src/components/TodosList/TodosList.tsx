import Todo from "../Todos/types/todo";
import TodosItem from "../TodosItem/TodosItem";
import styled from 'styled-components';
import React from "react";
import ITodo from "../Todos/types/todo";

const Ul = styled.ul<UlProps>`
display: flex;
flex-direction: column;
transition: all 500ms;
align-items: center;
transform: ${({$isVisible}) => $isVisible ? "translateY(0)" : "translateY(-200%)"};
`

interface UlProps {
    $isVisible: boolean;
}

interface TodosListProps {
    todos: ITodo[];
    isVisibleListTodos: boolean;
    onToggleComplited: (id: string) => void;
}

const TodosList: React.FC<TodosListProps> = ({todos, isVisibleListTodos, onToggleComplited}) => {
  return (
      <Ul $isVisible={isVisibleListTodos}>
          {todos.map((todo: Todo) => (
              <TodosItem key={todo.id} item={todo} onToggleComplited={onToggleComplited}/>
          ))}
      </Ul>
  )
}

export default TodosList;