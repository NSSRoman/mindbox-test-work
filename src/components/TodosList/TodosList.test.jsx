import { render, screen } from '@testing-library/react';
import TodosList from './TodosList';

const data = [
              {id: '1', text: 'test', isComplited: false},
              {id: '2', text: 'react', isComplited: false},
              {id: '3', text: 'jest', isComplited: false}
             ];

describe('TodosList component', () => {
    it('TodosList renders', () => {
        render(<TodosList todos={data}/>);

        expect(screen.getByRole('list')).toBeInTheDocument();
        expect(screen.getByText(/react/)).toBeInTheDocument();
    });

    it('TodosList snapshot', () => {
        const utils = render(<TodosList todos={data}/>);

        expect(utils).toMatchSnapshot();
    });
  });
  