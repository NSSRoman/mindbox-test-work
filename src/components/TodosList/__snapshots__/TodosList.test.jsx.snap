// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodosList component TodosList snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="sc-eDvSVe dzgMgr"
      >
        <li
          class="sc-bcXHqe iIXjuB"
        >
          <label
            class="sc-dkrFOg hBpAeX"
          >
            <input
              class="sc-hLBbgP iyUlKq"
              type="checkbox"
            />
            <span
              class="sc-gswNZR hFzkvj"
            >
              test
            </span>
          </label>
        </li>
        <li
          class="sc-bcXHqe iIXjuB"
        >
          <label
            class="sc-dkrFOg hBpAeX"
          >
            <input
              class="sc-hLBbgP iyUlKq"
              type="checkbox"
            />
            <span
              class="sc-gswNZR hFzkvj"
            >
              react
            </span>
          </label>
        </li>
        <li
          class="sc-bcXHqe iIXjuB"
        >
          <label
            class="sc-dkrFOg hBpAeX"
          >
            <input
              class="sc-hLBbgP iyUlKq"
              type="checkbox"
            />
            <span
              class="sc-gswNZR hFzkvj"
            >
              jest
            </span>
          </label>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="sc-eDvSVe dzgMgr"
    >
      <li
        class="sc-bcXHqe iIXjuB"
      >
        <label
          class="sc-dkrFOg hBpAeX"
        >
          <input
            class="sc-hLBbgP iyUlKq"
            type="checkbox"
          />
          <span
            class="sc-gswNZR hFzkvj"
          >
            test
          </span>
        </label>
      </li>
      <li
        class="sc-bcXHqe iIXjuB"
      >
        <label
          class="sc-dkrFOg hBpAeX"
        >
          <input
            class="sc-hLBbgP iyUlKq"
            type="checkbox"
          />
          <span
            class="sc-gswNZR hFzkvj"
          >
            react
          </span>
        </label>
      </li>
      <li
        class="sc-bcXHqe iIXjuB"
      >
        <label
          class="sc-dkrFOg hBpAeX"
        >
          <input
            class="sc-hLBbgP iyUlKq"
            type="checkbox"
          />
          <span
            class="sc-gswNZR hFzkvj"
          >
            jest
          </span>
        </label>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
